cpp/ps0005/p0005.cpp

```diff
@@ -27,24 +30,31 @@ int main() {
                 result = num1 / num2;
             } else {
                 cout << "Error: Division by zero" ;
-                return 1; // Exit the program
-            }
+                return 2; // Exit the program
+            } // 1/1
             break;
         case '%':
             if (int(num2)!= 0) {
-                result = int(num1) % int(num2);
+                if (int(num1) == num1 && int(num2) == num2))
+                    result = int(num1) % int(num2);
+                else
+                {
+                    cout << "cannot execute modulo operation for floating point numbers." << endl;
+                    return 4;
+                }
             } else {
                 cout << "Error: Division by zero" ;
-                return 1; 
-            }
+                return 2; 
+            } // 0.5/1
             break;
         default:
             cout << "Invalid operation!"  ;
-            return 1; 
-    }
+            return 1; // 1/1 
+    } // 0.5/1

```

Comment:
- [x] used `switch` and `if` structures in a right way to handle different cases
- [x] handled division operation `/`
- [ ] handled modulo operation `%`
- [x] handled invalid operator case
- [x] chose the right operation for each case